lib/services/elasticsearch.js

```diff
@@ -196,25 +194,38 @@ class ElasticSearch extends Service {
       scrollId
     };
 
-    const cacheKey = scrollCachePrefix + this._kuzzle.constructor.hash(
+    const cacheKey = SCROLL_CACHE_PREFIX + this._kuzzle.constructor.hash(
       esRequest.scrollId);
 
     debug('Scroll: %o', esRequest);
 
-    return this._kuzzle.cacheEngine.internal.exists(cacheKey)
-      .then(exists => {
-        if (exists === 0) {
-          throw errorsManager.get('unknown_scroll_id');
-        }
+    if (_scrollTTL) {
+      const scrollDuration = ms(_scrollTTL);
 
-        // ms(scroll) may return undefined if in microseconds or in nanoseconds
-        const ttl = ms(_scrollTTL) || ms(this._config.defaults.scrollTTL);
+      if (scrollDuration > ms(this._config.maxScrollDuration)) {
+        throw errorsManager.get('scroll_duration_too_great', _scrollTTL);
+      }
+    }
 
-        return this._kuzzle.cacheEngine.internal.pexpire(cacheKey, ttl);
-      })
-      .then(() => this._client.scroll(esRequest))
-      .then(({ body }) => this._formatSearchResult(body))
-      .catch(error => this._esWrapper.reject(error));
+    const exists = await this._kuzzle.cacheEngine.internal.exists(cacheKey);
+
+    if (exists === 0) {
+      throw errorsManager.get('unknown_scroll_id');
+    }
+
+    // ms(scroll) may return undefined if in microseconds or in nanoseconds
+    const ttl = ms(_scrollTTL) || ms(this._config.defaults.scrollTTL);
```

Comment:
I know this code was already there, but we should add an assert for this configuration too (and precompute the value).